### PR TITLE
chore(core): cleanup testing utilities

### PR DESCRIPTION
**Pull request by Wondertan**
Just a bunch of cleanups and some of them should fix some core-related flakeyness. More description in comments

### CODE REVIEW

File: core/testing.go [hunk 1]
```diff
@@ -1,40 +1,59 @@
...
+	protocol, ip := GetEndpoint(nd)
 	client, err := NewRemote(protocol, ip)
-	return remote, client, err
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := client.Stop()
+		require.NoError(t, err)
+	})
+	err = client.Start()
+	require.NoError(t, err)
```
Comment (Wondertan) [inline core/testing.go:50]: So the client is now started and stopped automatically. Previously the func name was mentioning Start of the Client but wasn't really doing so. Also this simplifies tests as they dont need to copy boileplate lifecycling code

**Comment by Wondertan** [comment]
The coverage decreased a bit due to skipped test. The why is commented there

**Comment by Wondertan** [comment]
image
❓ ❓ ❓ ❓ ❓

**Review by adlerjohn — APPROVED** [review]

**Comment by renaynay** [comment]
Nvm I saw a weird error on this PR previously

**Review by renaynay — APPROVED** [review]
lfg (utACK on whether this completely resolves flakeyness, but very useful regardless)

**Pull request merged**